execution/src/execution.rs

```diff
@@ -189,7 +189,7 @@ impl<R: ExecutionRpc> ExecutionClient<R> {
         index: usize,
     ) -> Result<Option<Transaction>> {
         let tx = payload.transactions[index].clone();
-        let tx_hash = H256::from_slice(&keccak256(tx));
+        let tx_hash = H256::from_slice(&keccak256(tx.to_vec()));
```

Comment:
Too many useless memory copy / clone.

References:
- [`pub type Transaction<N> = VariableList<u8, N>;`](https://github.com/sigp/lighthouse/blob/ea78336/consensus/types/src/execution_payload.rs#L9)
- [`impl<T, N: Unsigned> Deref for VariableList<T, N>`](https://docs.rs/eth2_ssz_types/latest/ssz_types/struct.VariableList.html#impl-Deref)
- [`pub fn keccak256<S>(bytes: S) -> [u8; 32]`](https://docs.rs/ethers/1.0.2/ethers/core/utils/fn.keccak256.html)
- [`pub struct H256(pub [u8; 32]);`](https://docs.rs/ethereum-types/0.14.1/ethereum_types/struct.H256.html)
- [`impl From<[u8; 32]> for H256`](https://docs.rs/ethereum-types/0.14.1/ethereum_types/struct.H256.html#impl-From%3C%5Bu8%3B%2032%5D%3E-for-H256)